flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/codegen/MatchCodeGenerator.scala

```diff
@@ -574,13 +584,14 @@ class MatchCodeGenerator(
   private def generateMeasurePatternVariableExp(patternName: String): GeneratedPatternList = {
     val listName = newName("patternEvents")
 
+    reusableMemberStatements.add(s"java.util.List $listName = new java.util.ArrayList();")
     val code = if (patternName == ALL_PATTERN_VARIABLE) {
       addReusablePatternNames()
 
       val patternTerm = newName("pattern")
 
       j"""
-         |java.util.List $listName = new java.util.ArrayList();
+         |$listName.clear();
```

Comment:
Do we have tests to cover this change? 

I would prefer `$listName = new java.util.ArrayList();`, because `$listName.clear();` will reuse the output object and may cause some unexpected problems (maybe `$listName` will be buffered in downstream).

Reply:
About `$listName = new java.util.ArrayList();` and things can be buffered downstream, that's a good point, I'll make the change.